library/network_connections.py

```diff
@@ -336,6 +336,7 @@ def ifcfg_create(
             ifcfg["ONBOOT"] = "no"
 
         ifcfg["DEVICE"] = connection["interface_name"]
+        ifcfg["MACADDR"] = connection["cloned_mac"]
 
```

Comment:
@thom311  are we still planning to support the new feature for the initscripts provider  ? IIRC,  we only do bugfix in the initscripts provider.